test/interactions/clickInteractionTests.ts

```diff
@@ -1,158 +1,208 @@
       });
 
-      it("can register multiple callbacks", () => {
-        clickInteraction.attachTo(component);
+      it("can register multiple onClick callbacks and unregister one", () => {
+        const callback1 = new TestClickCallback();
+        const call1 = () => callback1.call();
 
-        let callback1WasCalled = false;
-        let callback1 = () => callback1WasCalled = true;
+        const callback2 = new TestClickCallback();
+        const call2 = () => callback2.call();
 
-        let callback2WasCalled = false;
-        let callback2 = () => callback2WasCalled = true;
+        clickInteraction.onClick(call1);
+        clickInteraction.onClick(call2);
+        clickInteraction.offClick(call1);
+        clickPoint(testPoint);
 
-        clickInteraction.onClick(callback1);
-        clickInteraction.onClick(callback2);
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), 0, 0);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), 0, 0);
-        assert.isTrue(callback1WasCalled, "Click interaction should trigger the first callback");
-        assert.isTrue(callback2WasCalled, "Click interaction should trigger the second callback");
+        assert.isFalse(callback1.getCalled(), "Callback1 should be disconnected from the click interaction");
+        assert.isTrue(callback2.getCalled(), "Callback2 should still exist on the click interaction");
+      });
 
-        callback1WasCalled = false;
-        callback2WasCalled = false;
-        clickInteraction.offClick(callback1);
-        TestMethods.triggerFakeMouseEvent("mousedown", component.content(), 0, 0);
-        TestMethods.triggerFakeMouseEvent("mouseup", component.content(), 0, 0);
-        assert.isFalse(callback1WasCalled, "Callback1 should be disconnected from the click interaction");
-        assert.isTrue(callback2WasCalled, "Callback2 should still exist on the click interaction");
+      it("onClick returns this", () => {
+        const value = clickInteraction.onClick();
+        assert.strictEqual(value, clickInteraction);
+      });
 
-        svg.remove();
+      it("offClick returns this", () => {
+        const value = clickInteraction.offClick();
+        assert.strictEqual(value, clickInteraction);
       });
+    });
 
-      it("does not trigger callback when touch event is cancelled", () => {
-        let callbackCalled = false;
-        let callback = () => callbackCalled = true;
-        clickInteraction.onClick(callback);
+    describe("callbacks", () => {
+      let callback: TestClickCallback;
+
+      beforeEach(() => {
+        callback = new TestClickCallback();
+        clickInteraction.onClick((point: Plottable.Point) => callback.call(point));
+      });
+
+      [TestMethods.InteractionMode.Mouse, TestMethods.InteractionMode.Touch].forEach((mode: TestMethods.InteractionMode) => {
+        it("invokes onClick callback on single location click for " + TestMethods.InteractionMode[mode], () => {
```

Comment:
We prefer to use templating where possible.  No more " + "'s
it(`invokes onClick callback on single location click for ${TestMethods.InteractionMode[mode]}`,